src/video/macros.rs

```diff
@@ -27,9 +27,8 @@ macro_rules! impl_enum_frameintervals {
                 if ret.is_err() {
                     if v4l2_struct.index == 0 {
                         return Err(ret.err().unwrap());
-                    } else {
-                        return Ok(frameintervals);
                     }
+                    return Ok(frameintervals);
                 }
 
                 if let Ok(frame_interval) = FrameInterval::try_from(v4l2_struct) {
```

Comment:
Is this what you were thinking about? It's a little scary in the macro, but `cargo check` is happy!


```suggestion
                if let Err(e) = ret {
                    if v4l2_struct.index == 0 {
                        return Err(e);
                    }
                    return Ok(frameintervals);
                }

                if let Ok(frame_interval) = FrameInterval::try_from(v4l2_struct) {
```